build-system/tasks/performance-test/measure-documents.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fs = require('fs');
+const puppeteer = require('puppeteer');
+const {
+  CONTROL,
+  EXPERIMENT,
+  RESULTS_PATH,
+  urlToCachePath,
+} = require('./helpers');
+const {map} = require('bluebird');
+
+/**
+ * Setup measurement on page before navigating to the URL. Performance
+ * observers need to be initialized before content begins to load to take
+ * measurements.
+ *
+ * @param {Puppeteer.page} page
+ * @return {Promise} Resolves when script is evaluated
+ */
+const setupMeasurement = page =>
+  page.evaluateOnNewDocument(() => {
+    window.longTasks = [];
+    window.cumulativeLayoutShift = 0;
+    window.measureStarted = Date.now();
+    window.largestContentfulPaint = 0;
+
+    const longTaskObserver = new PerformanceObserver(list =>
```

Comment:
dumb question on my part, but is not possible (or maybe its more difficult?) to set this all up with a single observer?

Reply:
It should be possible, but then instead of three callbacks that do one thing, there would be one callback that does three things (with a switch or conditional). This seemed cleaner but I'm open to be persuaded otherwise!